profile/src/wallet_kit_common/secure_random_bytes.rs

```diff
@@ -0,0 +1,37 @@
+use rand::{rngs::OsRng, RngCore};
+
+/// Generates `N` random bytes using a cryptographically
+/// secure random generator and returns these bytes as
+/// a Vec<u8>.
+pub fn generate_bytes<const N: usize>() -> Vec<u8> {
+    let mut csprng = OsRng;
```

Comment:
I think `OsRng` was considered secure enough, but still, probably not as secure as letting SecureEnclave on an iPhone generate random bytes, why I think it should be possible for clients to inject the entropy for the first "Babylon" `DeviceFactorSource` (mnemonic) for new users.